app/src/main/java/com/greenfox/tribesoflagopusandroid/fragments/TroopsFragment.java

```diff
@@ -2,44 +2,56 @@
 
 
 import android.os.Bundle;
-import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.ListView;
 
 import com.greenfox.tribesoflagopusandroid.R;
+import com.greenfox.tribesoflagopusandroid.TribesApplication;
 import com.greenfox.tribesoflagopusandroid.adapter.TroopAdapter;
 import com.greenfox.tribesoflagopusandroid.api.model.gameobject.Troop;
+import com.greenfox.tribesoflagopusandroid.api.model.response.TroopsResponse;
+import com.greenfox.tribesoflagopusandroid.api.service.ApiService;
+import com.greenfox.tribesoflagopusandroid.api.service.MockApiService;
 
-import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class TroopsFragment extends Fragment {
 
     private TroopAdapter troopAdapter;
+    @Inject ApiService apiService;
+    List<Troop> troopArrayList;
+
 
     public TroopsFragment() {
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
+        TribesApplication.app().basicComponent().inject(this);
+
+        apiService.getTroops(1).enqueue(new Callback<TroopsResponse>() {
+            @Override
+            public void onResponse(Call<TroopsResponse> call, Response<TroopsResponse> response) {
+                troopArrayList = response.body().getTroops();
+            }
+
+            @Override
+            public void onFailure(Call<TroopsResponse> call, Throwable t) {
+
+            }
+        });
 
-        ArrayList<Troop> troopArrayList = new ArrayList<>();
-        Troop troop = new Troop(1,1,5,5,5);
-        Troop troop1 = new Troop(1,1,10,8,2);
-        Troop troop2 = new Troop(1,2,20,3,7);
-
-        troopAdapter = new TroopAdapter(this.getContext(), troopArrayList);
-        troopAdapter.add(troop);
-        troopAdapter.add(troop1);
-        troopAdapter.add(troop2);
-        troopAdapter.add(troop1);
-        troopAdapter.add(troop1);
-        troopAdapter.add(troop1);
-        troopAdapter.add(troop1);
-        troopAdapter.add(troop1);
+        troopAdapter = new TroopAdapter(getContext(), troopArrayList);
```

Comment:
@viktorhegyi here the `troopArrayList` is null, this will cause a NPE when using real server.
initialize the troop adapter with a new empty arraylist, and in the `onResponse` just add to the adapter (`.addAll()`)